### PR TITLE
Add install script for fail2ban

### PR DESCRIPTION
**Pull request by Landrash**
## Description:
Adds a installation script of the [fail2ban](https://github.com/fail2ban/fail2ban) service.

## Checklist (Required):
  - [x] The code change is tested and works locally.
  - [x] The code is compliant with [Contributing guidelines](https://github.com/home-assistant/hassbian-scripts/blob/master/.github/CONTRIBUTING.md)

### If pertinent:
  - [x] Script has validation check of the job.
  - [x] Created/Updated documentation at `/docs`

### CODE REVIEW

File: package/opt/hassbian/suites/fail2ban.sh [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+action = %(action_mwl)s
+
+[ha]
+enabled = true
+filter = ha
+logpath = /home/homeassistant/.homeassistant/home-assistant.log
+
+# 3600 seconds = 1 hour
+bantime = 3600
+bantime = 30 # during testing it is useful to have a short ban interval, comment out this line later
```
Comment (ludeeus) [inline package/opt/hassbian/suites/fail2ban.sh]: Does Fail2Ban allow duplicates in the configuration?
Or should one of these be commented out by default?

File: docs/fail2ban.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# Monitor
```
Comment (ludeeus) [inline docs/fail2ban.md]:
```suggestion
# Fail2Ban
```
Added suggestion for a title change.

**Review by ludeeus — COMMENTED** [review]
Added some minor comments, other than those it looks good :)

File: package/opt/hassbian/suites/fail2ban.sh (hunk 1 above)
Comment (ludeeus) [reply package/opt/hassbian/suites/fail2ban.sh]:
```suggestion
#bantime = 30 # during testing it is useful to have a short ban interval, comment out this line later
```
Tested the script now, the service fails if both are present.

File: package/opt/hassbian/suites/fail2ban.sh [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+# Maximum amount of login attempts before IP is blocked
+maxretry = 3" > "$FAIL2BANJAIL"
+fi
+
+echo "Restarting fail2ban service"
+systemctl restart fail2ban
+
+echo "Checking the installation..."
+validation=$(which -x fail2ban-client)
```
Comment (ludeeus) [inline package/opt/hassbian/suites/fail2ban.sh]: This results in `Illegal option -x`

```suggestion
validation=$(which fail2ban-client)
```

**Review by ludeeus — CHANGES_REQUESTED** [review]
Two minor changes :)

**Comment by Landrash** [comment]
Changes applied ☺️

**Review by ludeeus — APPROVED** [review]

**Pull request merged**